.circleci/config.yml

```diff
@@ -57,80 +48,12 @@ jobs:
epth 1 -maxdepth 1 -type d -exec basename {} \\;))\n\n# Get all packages that require a dedicated executor\ndedicated_executor_packages=()\nfor i in << pipeline.parameters.test-executor-packages >>; do dedicated_executor_packages+=($i); done\n\n# Validate parameter test-executors-count (which needs to be a parameter because it's used in the \"parallelism\" declaration)\nrequired_executors=\"${#dedicated_executor_packages[@]}\"\nif [[ $CIRCLE_NODE_TOTAL -ne $((required_executors+1)) ]]; then\n  echo \"The 'test-executors-count' must be 1 more than the number of packages in parameter 'test-executor-packages\".\n  exit 1\nfi\n\n# Get all other packages that don't require a dedicated executor - these will be run on a single executor\ncommon_executor_packages=()\nfor package in \"${all_packages[@]}\"; do\n  if [[ ! \" ${dedicated_executor_packages[@]} \" =~ \" ${package} \" ]]; then\n    common_executor_packages+=(\"$package\")\n  fi\ndone\n\n# Function to find test classes for a package and write them to file\nfind_test_classes() {\n  local package=$1\n  local output_file=$2\n\n  if [ -d \"uk/gov/justice/digital/hmpps/prisonerfromnomismigration/$package\" ]; then\n    local test_files=$(find \"uk/gov/justice/digital/hmpps/prisonerfromnomismigration/$package\" -name \"*.kt\" \\\n      | xargs -r grep -l -E '@Test|@ParameterizedTest' 2>/dev/null || true)\n    echo \"Found test_files for package=$package: $test_files\"\n    \n    # Convert test files to class names\n    if [ ! -z \"$test_files\" ]; then\n      echo \"$test_files\" \\\n        | sed 's@/@.@g' \\\n        | sed 's/\\.kt//g' \\\n        >> \"$output_file\"\n    fi\n  else\n    echo \"Directory uk/gov/justice/digital/hmpps/prisonerfromnomismigration/$package does not exist. Check parameter test-executor-packages.\"\n    exit 1\n  fi\n}\n\nindex_output_file=\"test_classes$CIRCLE_NODE_INDEX.txt\"\n> \"$index_output_file\"\n\nif [[ \"$CIRCLE_NODE_INDEX\" -lt \"$((CIRCLE_NODE_TOTAL-1))\" ]]; then\n  # Create a file containing the classes for this dedicated executor package\n  echo \"Finding test classes for dedicated executor from package=${dedicated_executor_packages[$CIRCLE_NODE_INDEX]}\"\n  find_test_classes \"${dedicated_executor_packages[$CIRCLE_NODE_INDEX]}\" \"$index_output_file\"\nelse\n  # Create a single file to contain test classes bundled from all other packages\n  echo \"Finding test classes for common executor from packages=${common_executor_packages[@]}\"\n  for common_executor_package in \"${common_executor_packages[@]}\"; do\n    find_test_classes \"$common_executor_package\" \"$index_output_file\"\n  done\nfi\n\necho \"Executor $CIRCLE_NODE_INDEX will run tests for classes: $(cat $index_output_file)\"\n\n# Fail the build if no test files were found - unless we're in the bundled executor (which could potentially have zero tests)\nif [[ \"$CIRCLE_NODE_INDEX\" -lt \"$((CIRCLE_NODE_TOTAL-1))\" ]]; then\n  [ -s test_classes$CIRCLE_NODE_INDEX.txt ] || circleci-agent step halt\nfi\n"
```

Comment:
I don't think is good - the code is hard enough to read with some nice formatting 😃